compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -516,17 +517,16 @@ public void visit(BLangBlockStmt block) {
 
     @Override
     public void visit(BLangVariableDef varDefNode) {
-        if (varDefNode.var.expr instanceof BLangRecordLiteral &&
-                ((BLangRecordLiteral) varDefNode.var.expr).type.tag == TypeTags.STREAM) {
-            ((BLangRecordLiteral) varDefNode.var.expr).name = varDefNode.var.name;
-        }
-
         varDefNode.var = rewrite(varDefNode.var, env);
         BLangVariable varNode = varDefNode.var;
 
         // Generate default init expression, if rhs expr is null
         if (varNode.expr == null) {
-            varNode.expr = getInitExpr(varNode.type);
+            if (varNode.type instanceof BStreamType) {
```

Comment:
Why do we need to special case streamType here? Put this logic inside the getInitExpr method.

Instanceof check is ugly.

Reply:
We had to special case the stream since we do want to get the identifier name as well as the corresponding BStream reference to deal with Siddhi. @MaryamZi , can u please check this.

Reply:
Variable names only used for debugging purposes. Why do you need to store variable name? We anyway store names in the balx

Reply:
I will merge for now. Lets get this fixed soon.

Reply:
Moved stream specific logic to the getInitExpr method with https://github.com/ballerina-platform/ballerina-lang/pull/6712.